code/modules/mob/living/brain/brain_item.dm

```diff
@@ -37,7 +37,7 @@
 		if(C.has_brain_worms())
 			var/mob/living/simple_animal/borer/B = C.has_brain_worms()
 			B.leave_victim() //Should remove borer if the brain is removed - RR
-		transfer_identity(C)
+	transfer_identity(C)
```

Comment:
Fun fact: according to brief tests, surgical brain removal takes the real brain from changelings on live. Dismemberment uses transfer_to_limb, which does explicitly check for changeling status and uses a fake brain if it's the case. I could add a fix for the first bit if desired.